### PR TITLE
Automated cherry pick of #75486: Fix fake clientsets in metrics.k8s.io.

### PR DESCRIPTION
**Pull request by misterikkit**
Cherry pick of #75486 on release-1.14.

#75486: Fix fake clientsets in metrics.k8s.io.

### CODE REVIEW

**Comment by tpepper** [comment]
/sig instrumentation
/sig scheduling
/kind bug
/release-note-none

@kubernetes/sig-instrumentation-pr-reviews and @kubernetes/sig-scheduling-pr-reviews please advise if this meets your threshold for critical bug fix backport to 1.14 branch.

**Comment by k82cn** [comment]
I'm ok with that, although I think it's not related to scheduling :)

**Comment by misterikkit** [comment]
> @kubernetes/sig-instrumentation-pr-reviews and @kubernetes/sig-scheduling-pr-reviews please advise if this meets your threshold for critical bug fix backport to 1.14 branch.

Can someone answer this question?

**Comment by brancz** [comment]
From instrumentation side it’s ok.

**Review by nikopen — APPROVED** [review]

**Comment by tpepper** [comment]
As per https://github.com/kubernetes/kubernetes/pull/75695#issuecomment-485998956 this is missing an approver still

**Comment by lavalamp** [comment]
What is the reason for cherrypicking this back a version?

**Comment by misterikkit** [comment]
The goal was to make testing possible on a downstream project which depends
on PodMetrics v1.14. IIRC we worked around the problem for now by
forking/wrapping the broken library. We will depend on 1.15 after it is
availabe, so if you feel this shouldn't be cherry-picked, I won't complain.

>

**Comment by lavalamp** [comment]
It doesn't seem like a critical bug fix, since it doesn't affect any server code and you can use the next client library for testing.

However it's also only a client testing fix, so it shouldn't make much difference for the release--very little risk.

/approve
/hold

@tpepper I think it's up to you to decide if this meets the bar or not.

**Comment by tpepper** [comment]
Looking at the code it seems like this has been broken for a very long time (since June of last year?), is a beta level feature, and a feature that is not proposed to go stable v1 in the 1.15 release.  Given that I feel it's lower criticality and I'm inclined to say the fix is in master and will be in 1.15 in a little over a month and that's sufficient.

That said it's also low risk and is now SIG approved.  Generally patch release defers to the SIG approvers.

:coin-flip:

**Comment by misterikkit** [comment]
Closing this since I think I've wasted enough of everyone's time. Thanks for considering it.

**Pull request closed without merging**
